src/server/cea_608_encoder/byte_pair_generator.py

```diff
@@ -135,18 +135,26 @@ def consume_captions(caption_list: list) -> dict:
     for caption in caption_list:
         if 'caption_id' not in caption:
             raise ValueError('A caption ID must be set for each caption')
+
+		# Setting the default background color to white if there is no background color passed in
+        foreground_color = 'white'
```

Comment:
This could be embedded in the method signature: https://github.com/capstone-team-a/Perception/blob/develop/src/server/cea_608_encoder/caption_string_utility.py#L156

Reply:
You still have to call the Function like this otherwise you have to set it as null, and then do it inside the function. If you do it here, all the defaults are located in the same area, as opposed to spreading them out to different functions.

Reply:
I see what you mean, but a midrow style in my eyes should be conditional (only if i want something other than the default), if we just want defaults, 608 is smart enough to automatically do it for us without us needing to give it bytes to do it. We are essentially wasting an operation by putting in defaults that 608 would do for us.

Reply:
Here's the issue. I can make it so we don't do a midrow style change if there is no underline and no color, but if you decide to change to change one, you have to set a value for the other. You have to go out of your way to add a several more lines for the same affect I've got here. I could add maybe 3-4 lines to make it so we don't call a midrow if both are null, but letting the defaults get set inside the midrow function adds way more lines, and makes it less readable than setting them here. I do like the idea of skipping a midrow if nothing is set, so I'll add those lines, but having the defaults set in the midrow function makes the logic wonky.

Reply:
ahh... this function is going to get really bloated with default values once preamble is added. I would prefer we have the loaded if/elses opposed to tons of default values in one function, but this code is still functioning as intended so i'll approve it

Reply:
Think about what it would take to put in a null color but a not a null underline. The if checks would be rampant. This function only takes two variables, if there's a function that takes more than two, it'll get even worse. Also this function already has to declare a variable as null, so why not save a lot of space and declare as default if you already have to declare as null.